src/components/CreateGroupForm/CreateGroupForm.tsx

```diff
@@ -0,0 +1,350 @@
+  { id: 14, name: 'Foreign Language' },
+  { id: 15, name: 'Health Education' },
+  { id: 16, name: 'Economics' },
+  { id: 17, name: 'Psychology' },
+  { id: 18, name: 'Sociology' },
+  { id: 19, name: 'Political Science' },
+  { id: 20, name: 'Philosophy' },
+];
+
+const INPUT_ERRORES_INITIAL = createGroupInputData.reduce(
+  (acc: { [key: string]: string }, { id }) => {
+    if (id !== 'days') {
+      acc[id] = '';
+    }
+    return acc;
+  },
+  {}
+);
+
+const inputErrorsReducer = (
+  state: typeof INPUT_ERRORES_INITIAL,
+  action: { type: string; payload: {} }
+): typeof INPUT_ERRORES_INITIAL => {
+  const { type, payload } = action;
+  switch (type) {
+    case 'UPDATE_ERRORS':
+      return { ...state, ...payload };
+
+    default:
+      throw Error('Type not defined');
+  }
+};
+
+const INPUT_VALUES_INITIAL = createGroupInputData.reduce(
+  (acc: { [key: string]: string | { id: number; name: string } }, { id }) => {
+    if (id === 'subject') {
+      acc[id] = { id: NaN, name: '' };
+    } else if (id === 'days') {
+      days.forEach((day) => {
+        acc[day] = 'off';
+      });
+    } else {
+      acc[id] = '';
+    }
+    return acc;
+  },
+  {}
+);
+
+const inputValuesReducer = (
+  state: typeof INPUT_VALUES_INITIAL,
+  action: { type: string; payload: {} }
+): typeof INPUT_VALUES_INITIAL => {
+  const { type, payload } = action;
+
+  switch (type) {
+    case 'UPDATE_VALUES':
+      return { ...state, ...payload };
+
+    default:
+      throw Error('Type not defined');
+  }
+};
+
+export default function CreateGroupForm() {
+  const [classes, setClasses] = useState<typeof query>([]);
+  // eslint-disable-next-line
+  const [filteredClasses, setFilteredClasses] = useState<typeof query>([]);
+
+  const [isListOpen, setIsListOpen] = useState(false);
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true);
+  const [isFormSubmitted, setIsFormSubmitted] = useState(false);
+
+  let timeOutId: NodeJS.Timeout;
+
+  // eslint-disable-next-line
+  const [inputValues, dispatch] = useReducer(
+    inputValuesReducer,
+    INPUT_VALUES_INITIAL
+  );
+
+  const [inputErrors, dispatchError] = useReducer(
+    inputErrorsReducer,
+    INPUT_ERRORES_INITIAL
+  );
+
+  const lastInputRef = useRef<HTMLInputElement>(null);
+  const submitRef = useRef<HTMLButtonElement>(null);
+  const autoComplete = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setClasses(query);
+    setFilteredClasses(query);
+  }, []);
+
+  const resetValues = () => {
+    dispatch({ type: 'UPDATE_VALUES', payload: INPUT_VALUES_INITIAL });
+    dispatchError({ type: 'UPDATE_ERRORS', payload: INPUT_ERRORES_INITIAL })
+  };
+
+  const checkFormValidation = () => {
+    const errors = Object.values(inputErrors);
+
+    for (let i = 0; i < errors.length; i += 1) {
+      if (errors[i] !== 'Hecho') {
+        setIsSubmitDisabled(true);
+        return;
+      }
+    }
+
+    if (
+      inputValues.L === 'off' &&
```

Comment:
Same comment about the array.every

Reply:
The inputValues object has more values than the day ones, so that's why i need to check them manually